crashlog/crashutils.c

```diff
@@ -164,13 +164,13 @@ unsigned long long get_uptime(int refresh, int *error)
 // Find system last kmsg from dropbox
 static int find_system_last_kmsg(char source[], int source_length) {
     struct dirent *entry;
-    DIR *dir = opendir(DROPBOX_DIR);
     int file_exist = 0;
 
     if (source == NULL) {
         LOGE("source is NULL.\n");
         return file_exist;
     }
+    DIR *dir = opendir(DROPBOX_DIR);
```

Comment:
why need to move opendir here?

Reply:
Variable dir going out of scope leaks the storage it points to at line 172.